lib/elixir/lib/port.ex

```diff
@@ -3,56 +3,91 @@ defmodule Port do
   Functions related to Erlang ports.
   """
 
+  @type name :: {:spawn, char_list | binary} |
+                {:spawn_driver, char_list | binary} |
+                {:spawn_executable, char_list | atom} |
+                {:fd, non_neg_integer, non_neg_integer}
+
   @doc """
-  See [`:erlang.open_port/2`](http://www.erlang.org/doc/man/erlang.html#open_port-2)
+  Opens an Erlang port given a tuple `name` and a list of settings.
+
+  ## Name
+
+  The supported values for `name` are:
+    * `{:spawn, command}` - to run an external program. The first space separated word of `command` will be considered as the name of the program to run, so use `{:spawn_executable, command}` to run a program having spaces in its name.
+    * `{:spawn_driver, command}` - similar to `{:spawn, command}`, but to run a loaded driver.
+    * `{:spawn_executable, filename}` - similar to `{:spawn, filename}`, but to run an external executable. With this option, `filename` in its whole is considered the name of the program to execute. 
+    * `{:fd, fd_in, fd_out}` - to access file descriptors used by Erlang, `fd_in` being used for standard input, `fd_out` for standard output.
+
+  For more information, see [`:erlang.open_port/2`](http://www.erlang.org/doc/man/erlang.html#open_port-2).
 
   Inlined by the compiler.
   """
+  @spec open(name, list) :: port
   def open(name, settings) do
     :erlang.open_port(name, settings)
   end
 
   @doc """
-  See [`:erlang.port_close/1`](http://www.erlang.org/doc/man/erlang.html#port_close-1)
+  Closes the `port`.
+
+  For more information, see [`:erlang.port_close/1`](http://www.erlang.org/doc/man/erlang.html#port_close-1).
 
   Inlined by the compiler.
   """
+  @spec close(port) :: true
   def close(port) do
     :erlang.port_close(port)
   end
 
   @doc """
-  See [`:erlang.port_command/2`](http://www.erlang.org/doc/man/erlang.html#port_command-2)
+  Sends `data` to the port driver `port`.
+
+  For more information, see [`:erlang.port_command/2`](http://www.erlang.org/doc/man/erlang.html#port_command-2).
 
   Inlined by the compiler.
   """
+  @spec command(port, iodata, options) :: true
```

Comment:
options is not a valid type, i think the correct would be `[:force | :nosuspend]`